app/services/services/services.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Services
+  module Services
+    include Errors
+
+    def self.deactivate(id)
+      service = Service.find id
+      raise Errors::PreconditionFailed unless service.approved?
+      service.inactive!
+      remove_from_algolia(service)
+    end
+
+    def self.remove_from_algolia(service)
```

Comment:
Maybe this can be factored out as a common utility since it is used in both services? Might be able to put it in `app/services/utils.rb`? Perhaps the error log message can be generalized as follows:

```ruby
def self.remove_from_algolia(record)
  record.remove_from_index!
rescue StandardError
  puts "failed to remove #{record.class} #{record.id} from algolia index"
end
```